lib/elixir/lib/enum.ex

```diff
@@ -803,11 +803,13 @@ defmodule Enum do
   @spec filter_map(t, (element -> as_boolean(term)),
     (element -> element)) :: list
 
-  def filter_map(enumerable, filter, mapper) when is_list(enumerable) do
+  def filter_map(enumerable, filter, mapper)
+  when is_list(enumerable) and is_function(filter, 1) and is_function(mapper, 1) do
```

Comment:
@lexmag :D

I would say though it should not be indented two-level, it should rather match the indentation of the inner element. For example:

```elixir
def foo(arg1, arg2)
    when ...

defp foo(arg1, arg2)
     when ...

defmacro foo(arg1, arg2)
         when ...
```

That's because it is just a regular call, as any other. It is for the reason we indent code like this:

```elixir
foo(bar, baz,
    bat, bad)
```